slick/src/main/scala/slick/jdbc/MySQLProfile.scala

```diff
@@ -94,6 +94,12 @@ trait MySQLProfile extends JdbcProfile { profile =>
         else l
       }
     }
+
+    //Reference: https://github.com/slick/slick/issues/1419
+    override def createTableNamer(meta: MTable): TableNamer = new TableNamer(meta){
+      override def schema = meta.name.catalog
+      override def catalog = meta.name.schema 
```

Comment:
Its only in MySQL. It seems to be mysql's jdbc driver causing issues. That would be out of scope 

I will submit the test shortly, it'd be very important in case the issue is resolved in the jdbc driver - that would cause alot of issues.